beta/plugins/remark-header-custom-ids.js

```diff
@@ -31,33 +31,51 @@ module.exports = ({
   return function transformer(tree) {
     visit(tree, 'heading', (node) => {
       const children = node.children;
+      const rawHeader = toString(node);
+      const match = /^.+(\s*\{.*\}\s*)$/u.exec(rawHeader);
+      /**
+       * we check if there is a heading and custom id with regex.
+       * If there is a match then we set the id to be the match[1] which is the custom id
```

Comment:
These comments are repeating the code. It's better if we use the code for things that aren't clear (like why does the code do things we want it to do).